### ISSUE TITLE
Option to specify train and validation split

### ISSUE THREAD

**Issue by eroell** [scverse/scvi-tools#1919]
**Is your feature request related to a problem? Please describe.**
When comparing models, specifying the same train-validation split for all models can be a comparison choice. It appears that in scvi-tools, the [DataSplitter](https://github.com/scverse/scvi-tools/blob/872ade25c521be0086231a134879a9c0c9b57022/scvi/dataloaders/_data_splitting.py#L56) class only allows specification of the `train_size` (or alternatively the `validation_size`).

**Describe the solution you'd like**
Being able to either
- Explicitly pass the train indices (or validation indices) as alternative to the `train_size` (or `validation_size`) during `model.train`
- Have an additional boolean argument `shuffle` which turns off np.random.RandomState's [permutation](https://github.com/scverse/scvi-tools/blob/872ade25c521be0086231a134879a9c0c9b57022/scvi/dataloaders/_data_splitting.py#L112) during `model.train`

would allow users to specify the train-validation split beforehand. The second option might be simpler, and would match the behaviour available in [sklearn.model_selection.train_test_split](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html)

Not sure if other people would also be interested in this or if it only leads to confusion - interested in what you think of it.

**Comment by adamgayoso** [comment]
yes we should add something like this. In the meantime a workaround is to pass completely held out adatas to the `model.get...` methods, which would cover the custom test set case

**Comment by adamgayoso** [comment]
@martinkim0 we finalized this right? Could you link the PR and close this? Thanks!

**Comment by martinkim0** [comment]
Closed via #2037, which implements the second proposed solution. It should be noted that the code currently takes the first `validation_size * n_obs` indices as the validation set and the next `train_size * n_obs` as the training set.
